enhancements/olm/olm-managed-operator-metrics.md

```diff
@@ -0,0 +1,145 @@
+
+#### Telemeter Requirements
+
+Once OLM is capable of managing `ServiceMonitor` and `PrometheusRule` objects along with the lifecycle of an operator, OLM must allow Red Hat owned operators to take advantage of Telemeter. Before moving forward, let's review the requirements that must be met for Telemeter to scrape an application's metrics:
+
+1. The application must be deployed in a namespace that is both prefixed with `openshift-` and labeled with `openshift.io/cluster-monitoring=true`
+2. A [ServiceMonitor object](https://github.com/coreos/prometheus-operator/blob/master/Documentation/user-guides/getting-started.md#related-resources) must be created in the namespace mentioned above and point to the application's metrics endpoint
+3. A role/rolebinding must be created that provides the Prometheus Operator ServiceAccount with the appropriate RBAC priviledges to discover the `ServiceMonitors` and `PrometheusRules` objects in the newly created namespace.
+
+Both requirements 1 and 2 are fully met by the changes proposed above leaving our team with the duty to fulfill requirement 3.
+
+#### Fulfilling Requirement 3: Updating the Prometheus Operator ServiceAccount
+
+In order for the Prometheus Operator to recognize take advantage of this new feature, its ServiceAccount must have the proper RBAC permissions to view events on the `ServiceMonitors` and `PrometheusRules` objects in the operator's namespace.
+
+On upstream clusters, there is no way for OLM to identify which Prometheus Operator ServiceAccount should be granted the RBAC priviledges. As such, the cluster admin will need to update the correct Prometheus Operator's OperatorGroup to include the correct namespace.
+
+On OpenShift cluster, OLM can simply update the default Prometheus instance with the correct RBAC priviledges when the `openshift.io/cluster-monitoring=true` label is present on the namespace.
+
+#### Risks and Mitigations
+
+It is possible that the Prometheus team will not allow OLM managed operators to use the default prometheus operator to record metrics. If they are concerned about OLM managed operators overloading the prometheus instance, one solution that addresses this concern includes creating a new operator that is responsible for adding the `openshift.io/cluster-monitoring=true` label to namespaces and for generating the Telemter RBAC. This operator would:
+
+- Be deployed in the `openshift-monitoring` namespace
+- Have a list of approved OLM managed operators that may report metrics to Telemeter
+- Watch for events on namespaces
+- If an event occurs on a namespace that starts with `openshift-` and the namespace only includes valid operators, add the `openshift.io/cluster-monitoring=true` annotation and create RBAC that allows the Telemeter service account to watch for events in the namespace.
+
+Alternatively, we could only add the `openshift.io/cluster-monitoring=true` on operators that originated from the `redhat-operators` and `certified-operators` CatalogSources.
```

Comment:
_Note: Continuing the conversation from https://github.com/operator-framework/enhancements/pull/4_
From @ecordell:

> Another alternative we discussed is having a signature on allowed operators

Reply:
@ecordell what do you mean by a signature?

Reply:
Included in the latest version of the PR as an alternative suggestion.